File: SofiForce.Web/ClientApp/src/app/core/Models/SearchModels/SalesOrderPromotionAllSearchModel.ts
import { BaseSearchModel } from "./baseSearchModel";

export interface SalesOrderPromotionAllSearchModel extends BaseSearchModel {
    promotionId: number;
    clientId: number;
    salesId: number;
    promotionCode: string;
    clientCode: string;
    invoiceCode: string;
    invoiceDateFrom: Date | null;
    invoiceDateTo: Date | null;
}

File: SofiForce.Models/Models/SearchModels/SalesOrderPromotionAllSearchModel.cs
using Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Models
{
    public class SalesOrderPromotionAllSearchModel : BaseSearchModel
    {
        public int PromotionId { get; set; }
        public int ClientId { get; set; }
        public double SalesId { get; set; }

        public string PromotionCode { get; set; }
        public string ClientCode { get; set; }
        public string InvoiceCode { get; set; }

        public DateTime? InvoiceDateFrom { get; set; }
        public DateTime? InvoiceDateTo { get; set; }

        

    }
}
